jupyterdrive/gdrive/gapi_utils.js

```diff
@@ -208,21 +224,37 @@ define(function(require) {
                     error.name = 'GapiError';
                     return Promise.reject(error);
                 }
-                return authorize(true);
+                return authorize(true, config);
             });
         }
     };
 
+    var _handle = {};
+    var _conf_prm = new Promise(function(resolve){
+        _handle.resolve = resolve;
+    })
+
+
+    var config = function(conf){
+      _handle.resolve(conf);
+    }
+
     /**
      * Promise fullfilled when gapi is loaded, and authorization is complete.
      */
-    var gapi_ready = load_gapi_1().then(load_gapi_2).then(authorize);
+    var load_gapi = load_gapi_1().then(load_gapi_2)
+    var gapi_ready = Promise.all([load_gapi, _conf_prm]).then(
+      function(resolve,reject){
```

Comment:
Yes, make sens. 

Probably due to some leftover experimentations.
-- 
M

Le 11 févr. 2015 à 17:53, Kester Tong <notifications@github.com> a écrit :

> In jupyterdrive/gdrive/gapi_utils.js:
> 
> >      /**
> >       * Promise fullfilled when gapi is loaded, and authorization is complete.
> >       */
> > -    var gapi_ready = load_gapi_1().then(load_gapi_2).then(authorize);
> > +    var load_gapi = load_gapi_1().then(load_gapi_2)
> > +    var gapi_ready = Promise.all([load_gapi, _conf_prm]).then(
> > +      function(resolve,reject){
> This should be function(args) or similar. resolve and reject are the args the the anonymous function in the promise function, and are callbacks used to resolve/reject the promise. But then takes just one argument, which is whatever the Promise was resolved with. The function is correct though, it's just the names and extra argument are confusing.
> 
> —
> Reply to this email directly or view it on GitHub.
>